### PR TITLE
Make slowmode timer more precise [2]

### PR DESCRIPTION
**Pull request by ozon671games1**
for issue #785

### CODE REVIEW

**Comment by ozon671games1** [comment]
Also, problem with self-signed certificate can be solved by adding ./test/test-cdn.betterttv.net.cert in browser's certificate store.

File: src/features/channel-state.js [hunk 1]
```diff
@@ -31,20 +31,22 @@ var resetCountDown = function() {
...
     bttv.chat.store.chatCountDown = setInterval(function() {
+        var currentTimestamp = Date.now();
+        var remainingTime = endTimestamp - currentTimestamp;
         var $chatButton = $(chatButton);
 
-        if (timer === 0) {
+        if (remainingTime <= 0) {
             resetCountDown();
             return;
+        } else {
```
Comment (night) [inline src/features/channel-state.js]: If you're returning in the above `if`, then the `else` block is superfluous. That's why it doesn't exist in the code you're editing.

File: src/features/channel-state.js [hunk 2]
```diff
@@ -31,20 +31,22 @@ var resetCountDown = function() {
 var initiateCountDown = function(length) {
     if (bttv.chat.store.chatCountDown) clearInterval(bttv.chat.store.chatCountDown);
 
-    var timer = length;
+    var startTimestamp = Date.now();
+    var endTimestamp = startTimestamp + (length * 1000);
```
Comment (night) [inline src/features/channel-state.js]: You're setting a variable and then using it one time (and never again) in the next line. Just use `Date.now()` in the next line.

File: src/features/channel-state.js [hunk 3]
```diff
@@ -31,20 +31,22 @@ var resetCountDown = function() {
...
 
     bttv.chat.store.chatCountDown = setInterval(function() {
+        var currentTimestamp = Date.now();
+        var remainingTime = endTimestamp - currentTimestamp;
```
Comment (night) [inline src/features/channel-state.js]: You're setting a variable and then using it one time (and never again) in the next line. Just use `Date.now()` in the next line.

**Comment by night** [comment]
Please address the commented lines by updating your pull request. Once updated we can test and merge it in.

**Comment by night** [comment]
Thanks

**Pull request merged**